### PR TITLE
Synchronize WF Editor Labels and Labels referenced in the WF Report

### PR DESCRIPTION
**Pull request by assuntad23**
Fix for #16864 
Relies on @jmchilton 's work on #17264 

Simply implementing the replaceLabel function when the label is changed.  

## How to test the changes?
(Select all options that apply)
- [X] Instructions for manual testing are as follows:
  1. Create a workflow with a label on one of the steps
  2. Go to edit the Report, and create an item in the report that references a label
  3. Navigate back to workflow steps, and change the label.
  4. Go back to the report and see that the label is changed. 

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/Workflow/Editor/Index.vue [hunk 1]
```diff
@@ -653,7 +654,10 @@ export default {
         },
         onLabel(nodeId, newLabel) {
             const step = { ...this.steps[nodeId], label: newLabel };
+            const oldLabel = this.steps[nodeId].label;
             this.onUpdateStep(step);
+            const newMarkdown = replaceLabel(this.markdownText, "step", oldLabel, newLabel);
```
Comment (jmchilton) [inline client/src/components/Workflow/Editor/Index.vue]: The report functionality distinguishes between inputs and steps - steps are assumed to be things associated with jobs and inputs are "Input Dataset", "Input Dataset Collection", or "Input Parameters". Can you dispatch on that in here and then test it?

I think we should inform users - maybe via a Toast style message that their markdown report has been updated if it has changed? Also we probably want to catch exceptions coming out of replace label and warn users if invalid markdown is preventing the update but still let them make the label change to the workflow.

We also need to handle output labels but that can be a follow up PR for sure.

Comment (assuntad23) [reply client/src/components/Workflow/Editor/Index.vue]: I'll add the Toast message and some exceptions handling of course. Great idea and it really eases my stress over changing something the user has created. Thanks!

@jmchilton But regarding the inputs vs steps, this seems to be working for Inputs, as well as steps. Unless I'm testing it incorrectly. But when I select any of the Jobs objects to insert in the report (Job Parameters, Job Metrics, etc.) and choose a label I've associated with an Input Dataset/Colletction. Then change that label, the label is changing in the report.  Is there a different option that I need to select in the Markdown Editor, that doesn't say step="name_of_label"?

Comment (jmchilton) [reply client/src/components/Workflow/Editor/Index.vue]: https://usegalaxy.org/u/jmchilton/w/unnamed-workflow-7

Screenshot 2024-01-22 at 10 19 54 AM


    ```galaxy
    history_dataset_display(input=input1)
    ```

Comment (jmchilton) [reply client/src/components/Workflow/Editor/Index.vue]:
```suggestion
            const stepType = this.steps[nodeId].type;
            const isInput = ["data_input", "data_collection_input", "parameter_input"].indexOf(stepType) >= 0;
            const labelType = isInput ? "input" : "step";
            const newMarkdown = replaceLabel(this.markdownText, labelType, oldLabel, newLabel);
```

**Comment by assuntad23** [comment]
no longer relevant, commits were cherry picked into new pr

**Pull request closed without merging**